### PR TITLE
Fix slicing from struct columns and accessing struct columns

### LINKED ISSUE #8718: [BUG] Incorrect slice on `struct` column.
**Describe the bug**
This bug is two fold. Firstly, `cpp_copying.slice()` generates an incorrect slice of the underlying `struct` column. Furthermore, the fields of the column after slicing is also incorrect (needs to be correctly repopulated).

**Steps/Code to reproduce bug**
```
In [15]: import cudf
In [16]: sr = cudf.Series([{"a": "Hello world"}, {"a": "String 2"}, {"a": "String 3"}])
In [17]: sr[1:]
Out[17]: 
1    {'0': 'Hello world'}
2       {'0': 'String 2'}
dtype: struct
In [18]: 
```

**Expected behavior**
```
1       {'a': 'String 2'}
2       {'a': 'String 3'}
dtype: struct
```

### PR DESCRIPTION
**Pull request by shaneding**
Fixes #8718

### CODE REVIEW

**Review by galipremsagar — APPROVED** [review]

File: python/cudf/cudf/tests/test_struct.py [hunk 1]
```diff
@@ -178,3 +178,57 @@ def test_dataframe_to_struct():
...
+    sr = cudf.Series(series)
+    if not end:
+        expected = cudf.Series(series[start:])
+        assert sr[start:].to_arrow() == expected.to_arrow()
+    elif not start:
+        expected = cudf.Series(series[:end])
+        assert sr[:end].to_arrow() == expected.to_arrow()
+    else:
+        expected = cudf.Series(series[start:end])
+        assert sr[start:end].to_arrow() == expected.to_arrow()
```
Comment (isVoid) [inline python/cudf/cudf/tests/test_struct.py:234]: We probably don't need to do specific things even if `start|end == None`:
```python
>>> l = [1, 2, 3, 4]
>>> l[None:2]
[1, 2]
>>> l[2:None]
[3, 4]
```

**Review by isVoid — COMMENTED** [review]
Another nit, feel free to ignore.

**Comment by galipremsagar** [comment]
@shaneding Looks like there are some `struct` related pytest failures, could you look into it?

**Comment by galipremsagar** [comment]
rerun tests

**Review by galipremsagar — APPROVED** [review]

**Comment by galipremsagar** [comment]
@gpucibot merge

**Pull request merged**